rdt/transformers/categorical.py

```diff
@@ -69,6 +69,24 @@ def _get_intervals(data):
         """
         data = data.fillna(np.nan)
         frequencies = data.value_counts(dropna=False)
+        augmented_frequencies = frequencies.to_frame()
+        sortable_column_name = f'sortable_{frequencies.name}'
+        column_name = frequencies.name or 0
+
+        def tie_breaker(element):
+            data_new_index = data.reset_index(drop=True)
+            if pd.isna(element):
+                data_is_na = data_new_index.isna()
+                return data_is_na.loc[data_is_na == 1].index[0]
+
+            return data_new_index.loc[data_new_index == element].index[0]
+
+        augmented_frequencies[sortable_column_name] = frequencies.index.map(tie_breaker)
+        augmented_frequencies = augmented_frequencies.sort_values(
+            [column_name, sortable_column_name],
+            ascending=[False, True]
+        )
+        sorted_frequencies = augmented_frequencies[column_name]
```

Comment:
The problem here is that `value_counts` is not deterministic. There fore, sometimes different values are assigned to different categories if they both occur the same amount. To fix this, I added a tie breaker based on the index of the first occurrence of that category. I couldn't use the category itself as a tie breaker because we allow for categories of different types, which can't be compared